app/components/works/edit/directory_component.html.erb

```diff
@@ -0,0 +1,13 @@
+<% if name.present? %>
+  <tr class="attached-files">
+    <td class="file-level" style="--depth: <%= depth %>rem" colspan="4">
+       <i class="fa fa-folder" aria-hidden="true"></i> <strong><%= name %></strong>
+    </td>
+  </tr>
+<% end %>
+<% children_files.each do |file| %>
+   <%= render Works::Edit::AttachedFileComponent.new(attached_file: file.attached_file, depth:, form:) %>
+<% end %>
+<% children_directories.each do |directory| %>
+   <%= render Works::Edit::DirectoryComponent.new(directory:, form:) %>
+<% end %>
```

Comment:
Maybe worth considering for follow-on work: rendering these components using ViewComponents collection rendering capability (vs. Ruby iteration): https://viewcomponent.org/guide/collections.html

I don't think this is blocking in the slightest.